File: SEM 3/OOPL/LAB2/q4.java
import java.util.Scanner;

class q4
{
    public static void main(String args[])
    {
        int i,j,m,n,p,q;
        Scanner sc = new Scanner(System.in);
        System.out.println("Enter Number of Rows of Matrix A");
        m=sc.nextInt();
        System.out.println("Enter Number of Columns of Matrix A");
        n=sc.nextInt();
        System.out.println("Enter Number of Rows of Matrix B");
        p=sc.nextInt();
        System.out.println("Enter Number of Columns of Matrix B");
        q=sc.nextInt();
        int A[][]=new int[m][n];
        int B[][]=new int[p][q];

        if(m==p&&n==q)
        {
            for(i=0;i<m;i++)
            {
                for(j=0;j<n;j++)
                {
                    System.out.println("Enter Element "+(i+1)+","+(j+1)+" of Mattix A");
                    A[i][j]=sc.nextInt();
                }
            }
            for(i=0;i<p;i++)
            {
                for(j=0;j<q;j++)
                {
                    System.out.println("Enter Element "+(i+1)+","+(j+1)+" of Mattix B");
                    B[i][j]=sc.nextInt();
                }
            }
            int C[][]=new int[m][n];
            for(i=0;i<m;i++)
            {
                for(j=0;j<n;j++)
                {
                    C[i][j]=A[i][j]+B[i][j];
                }
            }
            System.out.println("Matrix Sum is ");
            for(i=0;i<m;i++)
            {
                for(j=0;j<n;j++)
                {
                    System.out.print(C[i][j]+" ");
                }
                System.out.println("");

            }
        }
        else
        {
            System.out.println("Matrices can't be added");
        }

    }
}

File: SEM 3/DSAL/LAB2/q4.c
#include <stdio.h>
#include <stdlib.h>
void *read(int *h,int r,int c)
{
    int i,j;
    for(i=0;i<r;i++)
    {
        for(j=0;j<c;j++)
        {
            scanf("%d",(h + i*r + j));
        }
    }
}
void *display(int *h,int r,int c)
{
    int i,j;
    for(i=0;i<r;i++)
    {
        for(j=0;j<c;j++)
        {
            printf("%d \t",*(h + i*c + j));
        }
        printf("\n");
    }
}
void *Multiply(int *h,int *g,int *f,int r1,int c1,int c2)
{
    int i,j,k;
    for(i=0;i<r1;i++)
    {
        for(j=0;j<c2;j++)
        {
            *(f + i*r1 + j)=0;
            for(k=0;k<c1;k++)
            {
                *(f + i*r1 + j)=*(f + i*r1 + j) + (*(h + i*r1 + k))*(*(g + k*r1 + j));
            }
        }
    }
}

int main()
{
    int m,n,p,q;
    printf("Enter rows and columns for Matrix A : \t ");
    scanf("%d %d",&m,&n);
    printf("Enter rows and columns for Matrix B : \t ");
    scanf("%d %d",&p,&q);
    int *a=(int*)malloc(m*n*sizeof(int));
    int *b=(int*)malloc(p*q*sizeof(int));
    int *c=(int*)malloc(m*q*sizeof(int));
    printf("Enter Matrix A \n");
    read(a,m,n);
    printf("Enter Matrix B \n");
    read(b,p,q);
    printf("Matrix A \n");
    display(a,m,n);
    printf("Matrix B \n");
    display(b,p,q);
    if(n==p)
    {
        printf("Matrix after Multiplication is \n");
        Multiply(a,b,c,m,n,q);
        display(c,m,q);
    }
    else
    {
        printf("Matrices can't be Multiplied");
    }
    return 0;
}